ethexe/common/src/mock.rs

```diff
@@ -0,0 +1,497 @@
+// This file is part of Gear.
+//
+// Copyright (C) 2025 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+pub use tap::Tap;
+
+use crate::{
+    Address, Announce, AnnounceHash, BlockHeader, CodeBlobInfo, Digest, ProgramStates, Schedule,
+    SimpleBlockData,
+    db::*,
+    events::BlockEvent,
+    gear::{BatchCommitment, ChainCommitment, CodeCommitment, Message, StateTransition},
+};
+use alloc::{collections::BTreeMap, vec};
+use gear_core::code::{CodeMetadata, InstrumentedCode};
+use gprimitives::{CodeId, H256};
+use itertools::Itertools;
+use nonempty::{NonEmpty, nonempty};
+use std::collections::{BTreeSet, VecDeque};
+
+pub trait Mock<Args> {
+    fn mock(args: Args) -> Self;
+}
```

Comment:
I cannot see actual reason for trait. `mock` constructors can be used inside usual impls right in this file

Reply:
actually is better to use Arbitrary from proptest crate instead of Mock, but I'm not ready to doing so big work now. Maybe in future.

Reply:
It would be nice to have an issue for this

Reply:
https://github.com/gear-tech/gear/issues/4881